assignment/hyosungkim_proposal.md

```diff
@@ -0,0 +1,42 @@
+
+
+## Technologies used
+
+I will begin with the leaflet map, but ultimately plan to use Mapbox as the base for my map. I plan to use ajax calls for retrieving data, leaflet to include the functionality of adding and removing markers, turf to include the functionality of number of nearby rentals using a buffer, and underscore functions for iterating over rows of similar data.
+
+
+
+## Design spec
+
+#### User experience
+
+The main audience for my application is the Licenses and Inspection. Using this application will help them better prioritize where to conduct inspections given that there are a limited number of inspectors and a backlog of inspection units. Hopefully, this will also help them better identify where are the illegal rentals and in the bigger picture, help curb cases of illegal evictions.
+
+- Are there any website/application examples in the wild to which you can compare your final?
+
+#### Layouts and visual design
+
+The main component of the application will be a map of Philadelphia. I want to include a risk score slider so that users may be able to filter out high-risk, medium-risk and low-risk scored units. Because there are so many rental units, I want to include a functionality that aggregates the markers when the map is zoomed farther out. There will be a top bar with the title and a brief description of the application. I either plan to have a side bar or no side bar to display more details of each unit (number of bedrooms, license history, violation history, etc.) when clicked. With the additional buffer/count functionality, I want to include a checkbox to calculate the number of similar rentals in its vicinity when checked and not when not checked.
+
+
+## Anticipated difficulties
+
+The biggest challenge for this is the amount of data points to display on the map. My current data has over 300,000 points considered possible non-owner occupied rentals and I think that the processing for this will be very slow. A possible work around this issue is to narrow down on what I plan to display, for example, filtering out and displaying rentals with risk scores greater than x. I am still considering whether or not I will by default, filter the dataset or allow the user to input a range of risk scores to filter.
```

Comment:
You're exactly right that this is the central difficulty for an application of this kind. 300,000 points may well be too many, but you could perhaps aggregate at lower zoom levels and only display "full data" when zoomed in enough to naturally display fewer elements. This might be a candidate for using vector tiles (which we won't be covering in any detail I'm afraid, but which I'd be happy to give you pointers on if it comes to that)